Removes test_full_coinjoin

AdamISZ opened the pull request:
This is a stale test that we never got to a properly
working state, but we now have test/e2e-coinjoin-test.py
which does the same job more effectively, using the RPC-API.
Also removed this from parameters of run_tests script.

kristapsk reviewed (approved):
cr utACK eea4cbc4cb6bd78ae76b78b37f3b1a2044df110a

AdamISZ commented:
As per discussion in #1229, added in also the newer equivalent, `test/test_e2e_coinjoin.py`. With some edits.

This looks like it bumps up total test time by about 40 seconds. The edits were actually related: while this test was run isolated, I could override default fidelity bond time values to cut down wallet load time but that borked other tests (because the changes were global); changed to use the `monkeypatch` feature in pytest to localize those changes.

kristapsk reviewed (approved):
ACK 3b3bd3959780309ee53127d66bb50e6c22bd6c44. Test suite passes.
